### PR TITLE
Handle Pow in Commutator expansion.

### PR DESCRIPTION
**Pull request by versusvoid**
#### Brief description of what is fixed or changed
Handle expansion of commutators like
```
[A**2, C] = [A*A, C] -> A*[A, C] + [A, C]*A
```
but for arbitrary integer exponent.

#### Release Notes

* physics.quantum
   * Pow expressions handled in Commutator expansion

### CODE REVIEW

File: sympy/physics/quantum/tests/test_commutator.py [hunk 1]
```diff
@@ -25,9 +26,16 @@ def test_commutator_identities():
     assert Comm(A, B*C).expand(commutator=True) == Comm(A, B)*C + B*Comm(A, C)
     assert Comm(A*B, C*D).expand(commutator=True) == \
         A*C*Comm(B, D) + A*Comm(B, C)*D + C*Comm(A, D)*B + Comm(A, C)*D*B
+    assert Comm(A, B**2).expand(commutator=True) == Comm(A, B)*B + B*Comm(A, B)
+    assert Comm(A**2, C**2).expand(commutator=True) == \
+        Comm(A*B, C*D).expand(commutator=True).replace(B, A).replace(D, C)
```
Comment (oscarbenjamin) [inline sympy/physics/quantum/tests/test_commutator.py]: Can this test be changed so this is explicitly listed what is expected on the rhs?

**Comment by oscarbenjamin** [comment]
This looks good (although I don't know this subject well).

Sorry for the long delay getting back. Can you explain a little more what this PR does?

File: sympy/physics/quantum/tests/test_commutator.py (hunk 1 above)
Comment (versusvoid) [reply sympy/physics/quantum/tests/test_commutator.py]: Purpose was to show that results from power and multiplication are the same, but expression `A*A` gets automatically replaced with `Pow(A, 2)` so I had to use expand and replace.
I'll add second equality to final expression.

**Comment by versusvoid** [comment]
It helps solve some problems from quantum mechanics text books, e.g. with [angular momentum operator](https://en.wikipedia.org/wiki/Angular_momentum_operator#Commutation_relations_involving_vector_magnitude).

File: sympy/physics/quantum/tests/test_commutator.py (hunk 1 above)
Comment (oscarbenjamin) [reply sympy/physics/quantum/tests/test_commutator.py]: You can test that both things equal the same:
```
expected = ...
assert expr1 == expected
assert expr2 == expected
```

**Comment by oscarbenjamin** [comment]
Thanks for this!

**Pull request merged**